src/main/java/de/tum/in/www1/artemis/web/rest/StudentExamResource.java

```diff
@@ -71,7 +71,31 @@ public StudentExamResource(ExamAccessService examAccessService, StudentExamServi
     public ResponseEntity<StudentExam> getStudentExam(@PathVariable Long courseId, @PathVariable Long examId, @PathVariable Long studentExamId) {
         log.debug("REST request to get student exam : {}", studentExamId);
         Optional<ResponseEntity<StudentExam>> accessFailure = examAccessService.checkCourseAndExamAndStudentExamAccess(courseId, examId, studentExamId);
-        return accessFailure.orElseGet(() -> ResponseEntity.ok(studentExamService.findOneWithExercises(studentExamId)));
+
+        StudentExam studentExam = studentExamService.findOneWithExercises(studentExamId);
+        // we reload the quiz exercise because we also need the quiz questions and it is not possible to load them in a generic way with the entity graph used in
+        for (int i = 0; i < studentExam.getExercises().size(); i++) {
```

Comment:
If I see this correctly, some of this code is duplicated.
I suggest that you try to extract methods for the duplicated parts